src/features/identityCheck/components/AddressWithAutoCompletion.tsx

```diff
@@ -0,0 +1,127 @@
+import { t } from '@lingui/macro'
+import { useNavigation } from '@react-navigation/native'
+import debounce from 'lodash.debounce'
+import React, { useEffect, useRef, useState } from 'react'
+import { Keyboard, TouchableOpacity } from 'react-native'
+
+import { AddressOption } from 'features/identityCheck/atoms/AddressOption'
+import { CenteredTitle } from 'features/identityCheck/atoms/CenteredTitle'
+import { ModalContent } from 'features/identityCheck/atoms/ModalContent'
+import { PageWithHeader } from 'features/identityCheck/components/layout/PageWithHeader'
+import { useIdentityCheckContext } from 'features/identityCheck/context/IdentityCheckContextProvider'
+import { IdentityCheckError } from 'features/identityCheck/errors'
+import { UseNavigationType } from 'features/navigation/RootNavigator'
+import { eventMonitoring } from 'libs/monitoring'
+import { useAddresses } from 'libs/place'
+import { accessibilityAndTestId } from 'tests/utils'
+import { ButtonPrimary } from 'ui/components/buttons/ButtonPrimary'
+import { TextInput } from 'ui/components/inputs/TextInput'
+import { SNACK_BAR_TIME_OUT, useSnackBarContext } from 'ui/components/snackBar/SnackBarContext'
+import { Invalidate } from 'ui/svg/icons/Invalidate'
+import { Spacer } from 'ui/theme'
+import { ACTIVE_OPACITY } from 'ui/theme/colors'
+
+export const AddressWithAutoCompletion = () => {
+  const { showErrorSnackBar } = useSnackBarContext()
+  const { navigate } = useNavigation<UseNavigationType>()
+  const { dispatch } = useIdentityCheckContext()
+  const [addressQuery, setAddressQuery] = useState('')
+  const [debouncedAddress, setDebouncedAddress] = useState<string>(addressQuery)
+  const [selectedAddress, setSelectedAddress] = useState<string>('')
+  const debouncedSetAddress = useRef(debounce(setDebouncedAddress, 500)).current
+
+  // TODO (LucasBeneston, antoinewg): Récupérer cityCode / postalCode du context idcheck
+  const { data: addresses = [], isError } = useAddresses({
+    query: debouncedAddress,
+    cityCode: '',
+    postalCode: '',
+    limit: 10,
```

Comment:
C'est mieux
```suggestion
    limit: 20,
```
et il faut garder que les lieux-dits ou les adresses avec numéros en résultat